### PR TITLE
Add commit/exec integration test

### PR DESCRIPTION
**Pull request by jimmyxian**
Add commit/exec integration test
Signed-off-by: Xian Chaobo <xianchaobo@huawei.com>

### CODE REVIEW

File: test/integration/api.bats [hunk 1]
```diff
@@ -13,24 +13,39 @@ function teardown() {
 }
 
 @test "docker build" {
-      start_docker 3
```
Comment (aluzzardi) [inline test/integration/api.bats]: Watch out for tabs / spaces - I believe we are using tabs in `*.bats` files and your editor replaced them by spaces.

File: test/integration/api.bats [hunk 2]
```diff
@@ -13,24 +13,39 @@ function teardown() {
...
+
+	run docker_swarm run -d --name test_container busybox sleep 500
+	[ "$status" -eq 0 ]
+
+	# commit container
+	run docker_swarm commit test_container commit_image_busybox
+	[ "$status" -eq 0 ]
+	
+	# cluster refresh image during 30 seconds        
+	sleep 35
```
Comment (aluzzardi) [inline test/integration/api.bats]: I think that's a problem on Swarm that your integration test found.

We should force an image refresh after a commit (like we do for build).

File: test/integration/api.bats [hunk 3]
```diff
@@ -172,9 +198,24 @@ function teardown() {
...
+
+	run docker_swarm run -d --name test_container busybox sleep 500
+	[ "$status" -eq 0 ]
+
+	# rename container
+	run docker_swarm rename test_container rename_container
+	[ "$status" -eq 0 ]
+
+	# cluster refresh state of container during 30 seconds	
+	sleep 35
```
Comment (aluzzardi) [inline test/integration/api.bats]: Same here - we should refresh

File: test/integration/api.bats [hunk 4]
```diff
@@ -172,9 +198,24 @@ function teardown() {
 	skip
 }
 
-# FIXME
 @test "docker rename" {
```
Comment (aluzzardi) [inline test/integration/api.bats]: Actually, I think we don't really support renaming in Swarm (this is a new feature).

We should definitely upstate the `Store` with the new name.

File: test/integration/api.bats (hunk 2 above)
Comment (jimmyxian) [inline test/integration/api.bats]: Yeah, we should update the images after commit\pull\tag\load... I will check it and fix this problem.

File: test/integration/api.bats (hunk 4 above)
Comment (jimmyxian) [inline test/integration/api.bats]: OK

File: test/integration/api.bats (hunk 2 above)
Comment (jimmyxian) [inline test/integration/api.bats]: I submit a PR to force refresh images when commit/tag/pull #650

File: test/integration/api.bats (hunk 3 above)
Comment (jimmyxian) [inline test/integration/api.bats]: Actually, I think we should also force refresh container when rename, what do you think about this?

**Comment by jimmyxian** [comment]
@aluzzardi 
Removed rename test And use exec -i

**Comment by jimmyxian** [comment]
ping @aluzzardi

**Comment by aluzzardi** [comment]
LGTM

**Pull request merged**